### PR TITLE
Remove support for `include_broken_batteries` in control commands

### LINKED ISSUE #665: Remove support for `include_broken_batteries` in the PowerDistributor
### What's needed?

This was supposed to be for handling battery data outages, but where the batteries themselves are connected and usable.  But this has stopped being a issue, because of more stable bms firmware.

This also interferes with the safety features on the microgrid service.  So this can be removed.

There was only one downstream use-case for this feature - `gap discharge` - and that will also be removed.

### Proposed solution

_No response_

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shsms**
This removes

- the `include_broken_batteries` fields in request objects and from
  function parameter lists,
- the caching of battery data, that was necessary when trying to use
  broken batteries,
- corresponding tests

Closes #665

### CODE REVIEW

**Comment by shsms** [comment]
Based on #692,  only the last two commits are relevant.  Will rebase after #692 is merged.

**Comment by daniel-zullo-frequenz** [comment]
The commit `Remove support for include_broken_batteries in control commands` LGTM, I'd optionally suggest to add the reason (why) the include broken batteries feature is removed in the commit message as it is stated in the issue that this ticket is addressing (This is just a personal preference as I found it useful when tracking down changes in the git repo itself without the need to look it up in github)

**Comment by shsms** [comment]
> I'd optionally suggest to add the reason (why) the include broken batteries feature is removed in the commit message

true, done

**Review by daniel-zullo-frequenz — DISMISSED** [review]

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
   * Original methods `{set_power/charge/discharge}` are now replaced by `propose_{power/charge/discharge}`
   * The `propose_*` methods send power proposals to the `PowerManagingActor`, where it can be overridden by proposals from other actors.
   * They no longer have the `adjust_power` flag, because the `PowerManagingActor` will always adjust power to fit within the available bounds.
+  * They no longer have a `include_broken` parameter.  The feature has been removed.
```
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md]:
```suggestion
  * They no longer have a `include_broken_batteries` parameter.  The feature has been removed.
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
Sorry, I didn't spot it before but I think the release notes should mention `include_broken_batteries` instead as that is the parameter exposed to the user

**Comment by shsms** [comment]
> should mention include_broken_batteries instead as that is the parameter exposed to the user

fixed

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**